### ISSUE TITLE
roll_instances_when_required not behaving as expected for vmss when upgrade mode is set to manual

### ISSUE THREAD

**Issue by ungureanuvladvictor** [terraform-providers/terraform-provider-azurerm#6096]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
`Terraform v0.12.23`

### Affected Resource(s)

* `azurerm_linux_virtual_machine_scale_set `

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "=2.0.0"
  features {
    virtual_machine_scale_set {
      roll_instances_when_required = false
    }
  }
}
```

### Panic Output
No panics observed

### Expected Behavior
I'd expect the setting of the feature block to not roll my instances of the vmss.

### Actual Behavior
Instances of the vmss are upgraded by Terraform even though the features block should prevent it.

### Steps to Reproduce
Create a plain `azurerm_linux_virtual_machine_scale_set` resource with `upgrade_mode` set to `manual`. This instructs Azure to not upgrade the instances. Also set your providers block to use the `roll_instances_when_required` set to `false`. Then apply terraform to create the initial resources and after that change `custom_data`. You'll observe that after the Azure update calls finish Terraform will start to upgrade each instance of the vmss.

I was able to track it down to https://github.com/terraform-providers/terraform-provider-azurerm/blob/4f69a18291f3630d9e8e964e41b3c409f9cc704f/azurerm/internal/services/compute/virtual_machine_scale_set_update.go#L43-L60. `if upgradeMode == compute.Manual {` is the check that makes it upgrade instances. I am not sure if this is intended like this or just a typo in the if condition.

### Important Factoids
Running in normal cloud in eastus.

**Comment by Dmitry1987** [comment]
I believe the only option to avoid that is using lifecycle with "custom_data" specified, to ignore any update of a custom_data until there's a maintenance window (of a manual roll anyway) and then apply it without the lifecycle setting, so it'll roll them. 
This surprised me yesterday, luckily not on production :D

**Comment by matthias50** [comment]
It looks like this was fixed by https://github.com/terraform-providers/terraform-provider-azurerm/pull/7513. I had this issue myself but after upgrading to the latest azurerm, this issue went away.

**Comment by Dmitry1987** [comment]
Sounds good, just saw my version is a bit old, will try, thanks @matthias50 
```
Terraform v0.12.27
+ provider.azurerm v2.9.0
```

**Comment by tombuildsstuff** [comment]
Fixed by #7513 - thanks for the heads up @matthias50 :)
